superset/assets/src/query/buildQueryObject.ts

```diff
@@ -1,21 +1,64 @@
 import FormData, { getGranularity } from './FormData';
 import Metric, { Metrics } from './Metric';
 
-// TODO: fill out the rest of the query object
+const DTTM_ALIAS = '__timestamp';
+
+// TODO: fill out the rest of the query object: filter
 export interface QueryObject {
   granularity: string;
-  groupby?: string[];
-  metrics?: Metric[];
+  groupby: string[];
```

Comment:
Not all of these fields are required as this interface should accommodate all query objects (e.g. `groupby` is optional in the query object for word cloud). I think a good way to type this is to match a given field with whether it is required in the api signature on the server side.